src/os/posix/x11.rs

```diff
@@ -606,6 +604,28 @@ impl Window {
         }
     }
 
+    #[inline]
+    pub fn set_icon(&mut self, icon: Icon) {
+        // XChangeProperty
+        let net_string = CString::new("_NET_WM_ICON").unwrap();
```

Comment:
instead of doing this. Do `b"_NET_WM_ICON\0"` as that doesn't create a object on the heap.

Reply:
Good point, did that and fixed the errors in the workflow.
Also added an example in the `examples` folder since it might be weird to understand under Linux
but that can be removed of course.